tests/testthat/test_carp_clustering.R

```diff
@@ -63,3 +63,65 @@ test_that("Supplying percent and k gives error", {
 
   expect_error(clustering(carp_fit, k = 5, percent = 0.25))
 })
+
+
+test_that("CARP-returned sparsity pattern ends with full fusions",{
+  carp.fit <- CARP(presidential_speech)
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,ncol(carp.fit$carp.sol.path$v.zero.inds)]),
+    nrow(carp.fit$carp.sol.path$v.zero.inds)
+  )
+})
+
+test_that("CARPVIZ-returned sparsity pattern begins with no fusions",{
+  carp.fit <- CARP(presidential_speech,alg.type = 'carpviz')
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,1]),
+    0
+  )
+})
+test_that("CARPVIZ-returned sparsity pattern ends with full fusions",{
+  carp.fit <- CARP(presidential_speech,alg.type='carpviz')
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,ncol(carp.fit$carp.sol.path$v.zero.inds)]),
+    nrow(carp.fit$carp.sol.path$v.zero.inds)
+  )
+})
+
+
+
+
+
+test_that("CARP-returned sparsity pattern begins with no fusions (uniform weights)",{
+  weight_mat <-matrix(1,nrow=nrow(presidential_speech),ncol=nrow(presidential_speech))
+  carp.fit <- CARP(presidential_speech,weights = weight_mat)
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,1]),
+    0
+  )
+})
+test_that("CARP-returned sparsity pattern ends with full fusions (uniform weights)",{
+  weight_mat <-matrix(1,nrow=nrow(presidential_speech),ncol=nrow(presidential_speech))
+  carp.fit <- CARP(presidential_speech,weights = weight_mat)
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,ncol(carp.fit$carp.sol.path$v.zero.inds)]),
+    nrow(carp.fit$carp.sol.path$v.zero.inds)
+  )
+})
+
+test_that("CARPVIZ-returned sparsity pattern begins with no fusions (uniform weights)",{
+  weight_mat <-matrix(1,nrow=nrow(presidential_speech),ncol=nrow(presidential_speech))
+  carp.fit <- CARP(presidential_speech,alg.type = 'carpviz',weights = weight_mat)
+  expect_equal(
+    sum(carp.fit$carp.sol.path$v.zero.inds[,1]),
+    0
+  )
+})
+test_that("CARPVIZ-returned sparsity pattern ends with full fusions (uniform weights)",{
```

Comment:
As above - I think some of these are redundant since `carpviz` is the default algorithm. Did you mean to set `carp` and `t` explicitly earlier?